kavita: 0.8.5.11 -> 0.8.7

M0streng0 opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->
**kavita: 0.8.5.11 -> 0.8.7**

Updated Kavita to version 0.8.7 ([[release notes](https://github.com/Kareadita/Kavita/releases/tag/v0.8.7)](https://github.com/Kareadita/Kavita/releases/tag/v0.8.7)).

This update skips several intermediate versions (0.8.6, 0.8.6.1, 0.8.6.2) that contained significant refactoring work described by upstream as "Spring Cleaning." Due to these extensive code changes, I had to modify the existing `restore-migrations.diff` patch, as well as update the dependencies.

Also, the `locale` upstream issue is now fixed and merged, so I removed the patch.

**Question for maintainers:** Given the scope of upstream changes, I had to refactor the existing `restore-migrations.diff` patch to work with the new codebase. Could you please review whether my refactored patch looks correct?

@Misterio77 @nevivurn 

_Sorry for any mistake, first PR to Nixpkgs._

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [x] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

M0streng0 committed:
kavita: 0.8.5.11 -> 0.8.7

nevivurn reviewed (changes requested):
Thanks for the contribution!

wrt the restore-migrations patch, we only need to support NixOS users doing 24.11 -> 25.05 and a potential future 25.05 -> 25.11 upgrade.
Given that the last available Kavita version on 24.11 was v0.8.3.2 ([here](https://github.com/NixOS/nixpkgs/blob/24.11/pkgs/servers/web-apps/kavita/default.nix#L13)), we should drop the patches.

We should leave a comment mentioning that upstream has a tendency to drop old migrations, though.

M0streng0 commented:
Apparently I requested you a review! Dummy me... I was on the phone yesterday, and surely I misclicked.

I do agree with you. I also don't think we need the restore-migrations patch because there is a big version difference between the one we're updating to and the one in nixpkgs, and the tendency from Kavita's devs to refactor older code, which makes it harder and harder to move from an older version to the latest.

M0streng0 committed:
kavita: remove restore-migrations patch

M0streng0 commented:
As suggested by @nevivurn, I removed the restore-migrations patch since it's not needed for the 24.11 -> 25.05 upgrade path. I agree with this approach and added a comment explaining upstream's migration behavior for future reference.

nevivurn reviewed (approved):
Tested working on my installation, thanks!
